src/domain.py

```diff
@@ -19,60 +99,51 @@ def build_juju_pod_spec(app_name,
                         charm_config,
                         image_meta,
                         prometheus_server_details=None):
+
     external_labels = json.loads(charm_config['external-labels'])
     advertised_port = charm_config['advertised-port']
 
-    spec = {
-        'containers': [{
-            'name': app_name,
-            'imageDetails': {
-                'imagePath': image_meta.image_path,
-                'username': image_meta.repo_username,
-                'password': image_meta.repo_password
-            },
-            'ports': [{
-                'containerPort': advertised_port,
-                'protocol': 'TCP'
-            }],
-            'readinessProbe': {
-                'httpGet': {
-                    'path': '/-/ready',
-                    'port': advertised_port
-                },
-                'initialDelaySeconds': 10,
-                'timeoutSeconds': 30
-            },
-            'livenessProbe': {
-                'httpGet': {
-                    'path': '/-/healthy',
-                    'port': advertised_port
-                },
-                'initialDelaySeconds': 30,
-                'timeoutSeconds': 30
-            },
-            'files': [{
-                'name': 'config',
-                'mountPath': '/etc/prometheus',
-                'files': {
-                    'prometheus.yml': yaml.dump({
-                        'global': {
-                            'scrape_interval': '15s',
-                            'external_labels': external_labels
-                        },
-                        'scrape_configs': [{
-                            'job_name': 'prometheus',
-                            'scrape_interval': '5s',
-                            'static_configs': [{
-                                'targets': [
-                                    f'localhost:{advertised_port}'
-                                ]
-                            }]
-                        }]
-                    })
-                }
-            }]
+    # This is the first pass at enabling the addition of more metrics to
+    # prometheus. For now these other metrics are hard coded but future
+    # commits will allow for operations teams to add arbitrary metrics.
+
+    prometheus_config = PrometheusConfigFile(
+        global_opts={
+            'scrape_interval': '15s',
+            'external_labels': external_labels
+        }
+    )
+    prometheus_config.add_scrape_config({
+        'job_name': 'prometheus',
+        'scrape_interval': '5s',
+        'static_configs': [{
+            'targets': [
+                f'localhost:{advertised_port}'
+            ]
         }]
-    }
+    })
+    prometheus_config.add_scrape_config({
```

Comment:
I think, we should have a config option enabling or disabling the cluster monitoring. If this will be deployed on top of the K8s cluster without the Metrics server enabled - this will fail (probably, the charm should be in the "blocked" state then, but not in the "hook failed" error).

Reply:
Yes I was planning on making this a configurable option in the next commit to this PR. The idea of putting it in a `BlockedStatus` if the `metrics-server` is not available is a good idea!

Reply:
I've scrapped the idea of having a `BlockedStatus` since I found some documentation that `metrics-server` is really for internal use only. Anyway, my latest commit scrapes cAdvisor and other endpoints directly.